### PR TITLE
add 'clusterResources' field in NSTemplateSet

### PR DESCRIPTION
**Pull request by xcoulon**
result of https://github.com/codeready-toolchain/api/pull/117

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: deploy/templates/nstemplatetiers/namespaces/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+advanced-code: "5c0e0f2"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/namespaces/metadata.yaml]: I guess this is a zombi from another PR?

File: deploy/templates/nstemplatetiers/clusterresources/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+advanced: "5c0e0f2"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/clusterresources/metadata.yaml]: This one too.

File: deploy/templates/nstemplatetiers/namespaces/metadata.yaml (hunk 1 above)
Comment (xcoulon) [reply deploy/templates/nstemplatetiers/namespaces/metadata.yaml]: yes, let me remove these 2 `metadata.yaml` files. There was no reason to have them committed! 🤦‍♂

Comment (xcoulon) [reply deploy/templates/nstemplatetiers/namespaces/metadata.yaml]: removed in 9424627

File: deploy/templates/nstemplatetiers/clusterresources/metadata.yaml (hunk 2 above)
Comment (xcoulon) [reply deploy/templates/nstemplatetiers/clusterresources/metadata.yaml]: removed in 9424627

**Comment by xcoulon** [comment]
> It seems that you picked commits from another PR...
> You can just revert all the (three) commits from March 11 because no changes needed after making NSTempalteSet a pointer in API. It doesn't affect CRDs.
> Besides maybe updating the API dependency :)

@alexeykazakov  I've updated the PR with a single commit that resulted from the execution of `make generate` in codeready-toolchain/api.

**Pull request merged**